tests/test_completion.py

```diff
@@ -158,37 +168,125 @@ def test_ignores_text_edit(self):
+            # insertText is present, but we should prefer textEdit instead.
+            'insertText': 'const',
+            'sortText': '3f800000const',
+            'kind': 14,
+            # Note the extra space here. We should ignore this!
+            'label': ' const',
+            'filterText': 'const',
             'textEdit': {
+                'newText': 'const',
                 'range': {
-                    'end': {'character': 1, 'line': 0},
-                    'start': {'character': 1, 'line': 0}
-                },
-                'newText': 'what'
+                    # Replace the single character that triggered the completion request.
+                    'end': {'character': 13, 'line': 6},
+                    'start': {'character': 12, 'line': 6}
+                }
             }
         }
-
         handler = CompletionHandler(self.view)
         handler.last_location = 1
         handler.last_prefix = ""
         result = handler.format_completion(item)
-        self.assertEqual(len(result), 2)
-        self.assertEqual("$what", result[0])
-        self.assertEqual("\\$what", result[1])
-        # settings.complete_using_text_edit = False
+        self.assertEqual(result, ('const\t  Keyword', 'const'))
+
+    def test_text_edit1(self):
+        yield 100  # wait for file to be open
+        handler = CompletionHandler(self.view)
+        handler.last_location = 1
+        handler.last_prefix = ""
+        result = [handler.format_completion(item) for item in intelephense_completion_sample]
+        self.assertEqual(
+            result,
+            [
+                ('$x\t  mixed', '\\$x'),
+                ('$_ENV\t  array', '\\$_ENV'),
+                ('$php_errormsg\t  string', '\\$php_errormsg'),
+                ('$_FILES\t  array', '\\$_FILES'),
+                ('$GLOBALS\t  array', '\\$GLOBALS'),
+                ('$argc\t  int', '\\$argc'),
+                ('$argv\t  array', '\\$argv'),
+                ('$_GET\t  array', '\\$_GET'),
+                ('$HTTP_RAW_POST_DATA\t  string', '\\$HTTP_RAW_POST_DATA'),
+                ('$http_response_header\t  array', '\\$http_response_header'),
+                ('$_POST\t  array', '\\$_POST'),
+                ('$_REQUEST\t  array', '\\$_REQUEST'),
+                ('$_SERVER\t  array', '\\$_SERVER'),
+                ('$_SESSION\t  array', '\\$_SESSION'),
+                ('$_COOKIE\t  array', '\\$_COOKIE'),
+                ('$this\t  Variable', '\\$this')
+            ]
+        )
+
+    def test_text_edit2(self):
+        yield 100  # wait for file to be open
+        handler = CompletionHandler(self.view)
+        handler.last_location = 1
+        handler.last_prefix = ""
+        result = [handler.format_completion(item) for item in clangd_completion_sample]
+        # We should prefer textEdit over insertText. This test covers that.
+        self.assertEqual(
+            result,
+            [
+                ('argc\t  int', 'argc'),  # noqa: E501
```

Comment:
What is needed to get rid of the noqa exceptions?

Reply:
I added a handful of newlines where necessary.